printf.c

```diff
@@ -111,6 +111,45 @@
 #define FLAGS_PRECISION (1U << 10U)
 #define FLAGS_ADAPT_EXP (1U << 11U)
 
+// math constants
+#define BASE_2U    2U
+#define BASE_8U    8U
+#define BASE_10U  10U
+#define BASE_16U  16U
+#define SHIFT_52U 52U
+#define FLOATING_HALF 0.5
+#define FLOATING_ONE_AND_HALF 1.5
+#define FLOATING_1eminus4 1e-4
+#define FLOATING_1e6      1e6
+#define ONE_U      1U
+#define ONE_ULL    1ULL
+#define I_1      1
+#define I_2      2
+#define I_4U     4U
+#define I_5U     5U
+#define I_6      6
+#define I_9U     9U
+#define I_10     10
+#define I_14     14
+#define I_100      100
+#define I_1023     1023
+#define I_1023U    1023U
+#define I_1023ULL  1023ULL
+#define X_0x07FFU   0x07FFU
+#define X_0x07FFULL 0x07FFULL
+#define I_1000        1000
+#define I_10000       10000
+#define I_100000      100000
+#define I_1000000     1000000
+#define I_10000000    10000000
+#define I_100000000   100000000
+#define I_1000000000  1000000000
+#define FLOATING_0_1760912590558    0.1760912590558

```

Comment:
I suspect some of these are math constants that have a name?
Not having enormous luck finding explanation for these while googling...

However, I think in this case, since these are used in one and only one place in the code, it might make sense to define these closer to their use. If they are static constexpr values in the scope of the functions they are used in, they will not be initialized multiple times and will be treated pretty much like defines, except with type safety, and I guess there they can be named as coefficients for some log approximation or whatnot.

Reply:
I found good definitions for these math constants.  In my next update, they'll be constexpr with decent names.